spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -948,6 +950,17 @@ public static org.apache.spark.sql.catalyst.TableIdentifier toV1TableIdentifier(
     return org.apache.spark.sql.catalyst.TableIdentifier.apply(table, database);
   }
 
+  static String tableUUID(org.apache.iceberg.Table table) {
+    if (table instanceof HasTableOperations) {
+      TableOperations ops = ((HasTableOperations) table).operations();
+      return ops.current().uuid();
+    } else if (table instanceof BaseMetadataTable) {
+      return ((BaseMetadataTable) table).table().operations().current().uuid();
+    } else {
+      return null;
```

Comment:
I think this should probably throw an exception instead of returning null.

Reply:
Also instead of `tableUUID` maybe `baseTableUUID` sine that's what we're really getting.

Reply:
Done